add CONTRIBUTING.md (#2893)
deltachat/deltachat-android: CONTRIBUTING.md

If you do a PR fixing a bug or adding a feature,
please embrace the coding convention you see in the corresponding files,
so that the result fits well together.

Do not refactor or rename things in the same PR
to make the diff small and the PR easy to review.

Project language is Java.

By using [Delta Chat Core](https://github.com/deltachat/deltachat-core-rust)
there is already a strong separation between "UI" and "Model".
Further separations and abstraction layers are often not helpful
and only add more complexity.

Try to avoid premature optimisation
and complexity because it "may be needed in some future".
Usually, it is not.

Readable code is better than having some Java paradigms fulfilled.
Classic Java has a strong drive to add lots of classes, factories, one-liner-functions.
Try to not follow these patterns and keep things really on point and simple.
If this gets in conflict with embracing existing style, however,
consistency with existing code is more important.

The "Delta Chat Core" is a high-level interface to what the UI actually needs,
data should be served in a form that the UI do not need much additional work.

> If this is not the case, one may consider to add a feature to "Delta Chat Core" or ask for that.

Author: "adding a feature request" or "asking for that" seem to be mostly the same.

what about:

```suggestion
If this is not the case, consider a feature proposal to "Delta Chat Core".
```
